File: backend/src/main/java/com/nomad/dto/VoiceChatRequest.java
package com.nomad.dto;

import java.util.List;
import java.util.Map;

public record VoiceChatRequest(
    String message,
    Double userLat,
    Double userLng,
    String selectedCategory,
    List<Map<String, Object>> nearbyPois,
    List<Map<String, String>> conversationHistory  // Para mantener contexto
) {}

File: app-android/app/src/main/java/com/nomad/app/data/dto/VoiceChatRequest.kt
package com.nomad.app.data.dto

import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable

@Serializable
data class VoiceChatRequest(
    val message: String,
    val userLat: Double? = null,
    val userLng: Double? = null,
    val selectedCategory: String? = null,
    val nearbyPois: List<@Contextual Map<String, @Contextual Any>>? = null,
    val conversationHistory: List<ConversationMessage>? = null
)

@Serializable
data class ConversationMessage(
    val role: String,  // "user" o "assistant"
    val content: String
)
